Engine/src/Engine/InputSystem.h

```diff
@@ -0,0 +1,118 @@
+#pragma once
+#include <map>
+#include <mutex>
+#include "GLFW/glfw3.h"
+#include "glm/vec2.hpp"
+
+namespace Engine {
+	/// <summary>

```

Comment:
Not necessarily against xml style documentation, but it feels inconsistent if this is the only place we have it in the project. Just some thoughts, we could look at standardizing comments later on